### PR TITLE
Use pythia8 EvtGen plugin in Pythia8Interface

### PR DESCRIPTION
**Pull request by mkirsano**

### CODE REVIEW

**Comment by davidlange6** [comment]
@cmsbuild, please test

**Comment by covarell** [comment]
Thanks for a quick feedback.

While for testing this is already OK, it seems that for the moment no configuration is possible of EvtGen data files. While for DECAY2010.DEC files and evt.pdl this may be fine at the moment, it is mandatory to make the name of .dec file (user decay file) configurable. Even better, it should be possible either to provide a FileInPath or a very long string to be parsed, containing the full dec file.

**Comment by bendavid** [comment]
For inlining a long string for DEC file, I would suggest to use the same solution which has been implemented for the SLHA tables:

https://github.com/cms-sw/cmssw/blob/CMSSW_7_1_18/GeneratorInterface/Pythia8Interface/plugins/Pythia8Hadronizer.cc#L200

**Comment by mkirsano** [comment]
Yes, I was thinking about a pair of iterations, this being for some preliminary testing, specifying user dec file with a symbolic link. Anyway, to fully use this we need a new pythia8 version.

**Comment by covarell** [comment]
+1

File: GeneratorInterface/Pythia8Interface/interface/Py8InterfaceBase.h [hunk 1]
```diff
@@ -51,6 +53,13 @@ namespace gen {
 	 unsigned int                   maxEventsToPrint;
          HepMC::IO_AsciiParticles*      ascii_io;
 
+         // EvtGen plugin
+         //
+         bool useEvtGen;
+         EvtGenDecays* evtgenDecays;
+         string evtgenDecFile;
```
Comment (davidlange6) [inline GeneratorInterface/Pythia8Interface/interface/Py8InterfaceBase.h]: @mkirsano 
 place add the std:: namespace to the string declarations, as eventually the person whose header you include  will fix their code and remove the using std from their header..

Comment (bendavid) [inline GeneratorInterface/Pythia8Interface/interface/Py8InterfaceBase.h]: Sorry, which header has "using std" ?

Comment (davidlange6) [inline GeneratorInterface/Pythia8Interface/interface/Py8InterfaceBase.h]: no idea. but it must be the case for this to compile, no? [I’m not suggesting to fix the header, just to protect against it eventually being fixed]

**Comment by mkirsano** [comment]
There is certainly "using namespace std" in pythia8 headers. I don't think that it can disappear, but probably is has sense to make all string declarations similar.

**Comment by davidlange6** [comment]
thanks!

**Pull request merged**